agent/recorder.go

```diff
@@ -92,13 +96,17 @@ func (r *SpanRecorder) loop() error {
 					}
 				}
 				cTime = time.Now()
-				err := r.SendSpans()
+				err, shouldExit := r.SendSpans()
 				if err != nil {
 					r.logger.Printf("error sending spans: %v\n", err)
 				}
+				if shouldExit {
+					ticker.Stop()
+					return nil
```

Comment:
I think we have to call `r.t.Kill(err)` here to make the tomb be marked as dead

Reply:
yes, we have to, I'm gonna make the change